OpenDreamRuntime/Procs/AsyncNativeProc.cs

```diff
@@ -96,23 +96,26 @@ protected override ProcStatus InternalResume()
                     }, Thread.Runtime.TaskScheduler);
                 }
 
-                // We need to call a proc
-                if (_callProcNotify != null) {
-                    var callProcNotify = _callProcNotify;
-                    _callProcNotify = null;
-
-                    Thread.PushProcState(callProcNotify);
-                    return ProcStatus.Called;
-                }
+                // We need to call a proc.
+                while (_callProcNotify != null || _callResult != null)
```

Comment:
^ I also added this loop as AsyncNativeProcs with multiple (synchronous) external calls in a row would deadlock.